src/rpc/masternode.cpp

```diff
@@ -107,6 +108,7 @@ UniValue initmasternode(const JSONRPCRequest& request)
         return "success";
     }
     // legacy
+    std::string _strMasterNodeAddr = request.params[1].get_str();
```

Comment:
we could add an error if the second parameter (IP address) is not provided for legacy masternodes